app/currency/management/commands/parse_privatbank_archive.py

```diff
@@ -1,20 +1,191 @@
-from django.core.management.base import BaseCommand, CommandError
-from currency.models import Rate
+import datetime
+import time
+import requests
+from requests.exceptions import HTTPError, ConnectionError
+
+from django.core.management.base import BaseCommand
+
+from currency.utils import to_decimal, check_exist_and_create_rate
+from currency import consts
+from currency import model_choices as mch
+from currency.models import Rate, Source
 
 
 class Command(BaseCommand):
-    # help = 'Get rate by rate_id and print to console'
 
-    def add_arguments(self, parser):
-        parser.add_argument('rate_ids', nargs='+', type=int)
+    # ===========PARSER SETTINGS=================
+
+    # path and name log file
+    log_file_name = 'app/currency/management/commands/parse_privatbank_archive_log.txt'
+
+    # begin from date, comment/uncomment what do you need
+    # start_date = datetime.date(2014, 7, 22)
+    start_date = datetime.date.today()
+
+    # number of retries when there is no connection
+    connection_retries = 20  # set number of attempts
+
+    # number of retries when response.status_code in [429, 500, 502, 503, 504]
+    response_retries = 10  # set number of attempts
+
+    # number of attempts (steps) back by archive dates, if response_data['exchangeRate'] is empty
+    response_data_retries = 10  # set number of attempts
+
+    # delay in seconds after each parsing iteration
+    delay_sec = 3
+
+    # maximum number of parser iterations (useful when testing)
+    max_parser_iterations = 10000
+
+    def add_line_to_log(self, line_text: str, write_mode: str = "a", out_to_terminal: bool = False):
+        """
+        function create log-file and write row to end
+        :param line_text: text that will be displayed in the log and in the terminal
+        :param write_mode: if "w" - rewrite file, if "a" - append row to end of file (by default = "a")
+        :param out_to_terminal: show message to terminal (by default = False)
+        :return:
+        """
+        f = open(self.log_file_name, write_mode)
+        f.write(f"{datetime.datetime.utcnow()} - {line_text}\n")
+        f.close()
+        if out_to_terminal:
+            self.stdout.write(line_text)
 
     def handle(self, *args, **options):
-        for rate_id in options['rate_ids']:
+
+        # get or create Source
+        source_name = 'PrivatBank'
+        source_url = 'https://api.privatbank.ua/p24api/pubinfo?exchange&json&coursid=11'
```

Comment:
https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls